common/src/main/java/com/wynntils/features/user/QuickCastFeature.java

```diff
@@ -0,0 +1,145 @@
+    private static int packetCountdown = 0;
+
+    private static void castFirstSpell() {
+        tryCastSpell(RIGHT, LEFT, RIGHT);
+    }
+
+    private static void castSecondSpell() {
+        tryCastSpell(RIGHT, RIGHT, RIGHT);
+    }
+
+    private static void castThirdSpell() {
+        tryCastSpell(RIGHT, LEFT, LEFT);
+    }
+
+    private static void castFourthSpell() {
+        tryCastSpell(RIGHT, RIGHT, LEFT);
+    }
+
+    private static void tryCastSpell(boolean a, boolean b, boolean c) {
+        ItemStack heldItem = McUtils.player().getItemInHand(InteractionHand.MAIN_HAND);
+
+        if (!WynnItemMatchers.isWeapon(heldItem)) {
+            McUtils.sendMessageToClient(
+                    new TextComponent("You can only quick-cast with a weapon!").withStyle(ChatFormatting.RED));
+            return;
+        }
+
+        List<String> loreLines = ItemUtils.getLore(heldItem);
+
+        boolean isArcher = false;
+        for (String lore : loreLines) {
+            Matcher matcher = INCORRECT_CLASS_PATTERN.matcher(lore);
+            if (matcher.matches()) {
+                McUtils.sendMessageToClient(new TextComponent("You can't use " + matcher.group(1) + "-type weapons!")
+                        .withStyle(ChatFormatting.RED));
+                return;
+            }
+            if (lore.contains("Archer/Hunter")) isArcher = true;
+        }
+
+        for (String lore : loreLines) {
+            Matcher matcher = LVL_MIN_NOT_REACHED_PATTERN.matcher(lore);
+            if (matcher.matches()) {
+                McUtils.sendMessageToClient(new TextComponent("You need to reach " + matcher.group(1) + " "
+                                + matcher.group(2) + " before using this weapon.")
+                        .withStyle(ChatFormatting.RED));
+                return;
+            }
+        }
+
+        boolean isSpellInverted = isArcher;
+        List<Boolean> spell = Stream.of(a, b, c).map(x -> isSpellInverted != x).toList();
+        boolean[] partialSpell = SpellManager.getLastSpell();
+        for (int i = 0; i < partialSpell.length; ++i) {
+            if (partialSpell[i] != spell.get(i)) {
+                McUtils.sendMessageToClient(new TextComponent("Please wait for the incompatible spell-cast to finish")
+                        .withStyle(ChatFormatting.RED));
+                return;
+            }
+        }
+        lastSelectedSlot = McUtils.inventory().selected;
+        List<Boolean> remainder = new ArrayList<>(spell.subList(partialSpell.length, spell.size()));
+        remainder.stream().map(x -> x ? RIGHT_CLICK_PACKET : LEFT_CLICK_PACKET).forEach(SPELL_PACKET_QUEUE::add);
+    }
+
+    @SubscribeEvent
+    public void onTick(ClientTickEvent e) {
+        if (!WynnUtils.onWorld() || e.getTickPhase() != ClientTickEvent.Phase.END) return;
+        if (SPELL_PACKET_QUEUE.isEmpty()) return;
+        if (--packetCountdown > 0) return;
+        if (McUtils.inventory().selected != lastSelectedSlot)
```

Comment:
Please don't use multi-line if statements without braces.

Reply:
Yeah, sorry, the spotless formatting git-hook sometimes wraps when I don't want it to.